lib/index.js

```diff
@@ -508,9 +508,14 @@ async function run(options = {}) {
         const packages = await previousPromise
         // copy object to prevent share .ncurc options between different packageFile, to prevent unpredictable behavior
         const rcResult = getNcurc({ packageFile })
+        const rcConfig = rcResult && rcResult.config ? rcResult.config : {}
+        if (options.recursive && options.reject) {
+          // inherit parent config options
+          rcConfig.reject = [...new Set([...options.reject, ...rcConfig.reject || []])]
```

Comment:
We should apply config inheritance to all options, not just reject.

Reply:
Yep. Can you please advise where to find the complete list of nested structured properties?

Reply:
All of the options are listed in [https://github.com/raineorshine/npm-check-updates/blob/main/lib/cli-options.js](cli-options.js). I hope you’re not planning on applying them individually?

Reply:
For merge we can apply such simple generic function with test coverage:
```javascript
function mergeArrays(arr1, arr2){
  return [...new Set([...(arr1 || []), ...(arr2 || [])])]
}

function mergeOptions(options1, options2){
  options1 = options1 || {}
  options2 = options2 || {}  
  const result = {...options1, ...options2 }
  for (const key in result) {
    if (result.hasOwnProperty(key)) {
      if (Array.isArray(options1[key]) && Array.isArray(options2[key])){
        result[key] = mergeArrays(options1[key], options2[key]) 
      }
    }
  }
  return result;
}
```

i.e. it is a normal merge, but if both properties are arrays, then we merge them also.
WDT?

Reply:
Sure, I'm okay with that.